modules/mini-games/tightspot.lua

```diff
@@ -538,166 +580,177 @@ local function insideBox(box, pos)
     local all_items = inv.get_contents()
     left_players[player.name] = all_items
 
+    -- Teleport the player to nauvis
     player.set_controller {type = defines.controllers.god}
     player.create_character()
     player.teleport({-35, 55}, "nauvis")
-    --gui
+
+    -- Hide the game gui
     local Main_gui = Gui.get_left_element(player, game_gui)
+    Gui.toggle_left_element(player, game_gui, false)
+
     local table = Main_gui.container["slider"].table
     table.visible = false
-    Gui.toggle_left_element(player, game_gui, false)
 end
 
+--- Triggered when a player joins the game
 local function player_join(event)
     local player = game.players[event.player_index]
     player.character.destroy()
     local items = left_players[player.name]
     local center = centers[player.name]
-    if items then
+    if items and not started[1] then
         player.teleport(center, variables.level.surface)
-        if started[1] then
-            player.set_controller {type = defines.controllers.spectator}
-        else
-            player.set_controller {type = defines.controllers.god}
-            Gui.toggle_left_element(player, game_gui, true)
-            for item, amount in pairs(items) do
-                player.insert {name = item, count = amount}
-            end
+        player.set_controller {type = defines.controllers.god}
+        Gui.toggle_left_element(player, game_gui, true)
+        for item, amount in pairs(items) do
+            player.insert {name = item, count = amount}
         end
     else
-        player.set_controller {type = defines.controllers.spectator}
+        player.set_controller{ type = defines.controllers.spectator }
         player.teleport(variables.level.center, variables.level.surface)
     end
 end
 
---gui
-local speed_slider
-local function speed_change(_, element, _)
-    if Mini_games.get_running_game() == "Tight_spot" then
-        game.speed = element.slider_value
-    end
-end
-
---game gui
+--- When the players balance changes update the gui
+Store.watch(balances, function(value, key, _)
+    local player = game.players[key]
+    local Main_gui = Gui.get_left_element(player, game_gui)
+    local table = Main_gui.container["Money"].table
+    table["Balance"].caption = value
+end)
 
-speed_slider =
-    Gui.element {
+--- Used to change the game speed using a slider
+local speed_slider =
+Gui.element {
     type = "slider",
     minimum_value = 1,
     maximum_value = 64,
     value = 4,
     value_step = 1
-}:on_value_changed(speed_change)
-local label_func =
-    Gui.element(
-    function(_, parent, name, caption1, caption2)
-        parent.add {
-            type = "label",
-            caption = caption1
-        }
-        parent.add {
-            type = "label",
-            caption = caption2,
-            name = name
-        }
+}
+:on_value_changed(function(_, element, _)
+    if Mini_games.get_running_game() == "Tight_spot" then
+        game.speed = element.slider_value
```

Comment:
mabye this shood update the slider for all other players?

Reply:
Was an issue with the src. Will add the correction.